addons/source-python/packages/source-python/players/engines/orangebox/tf.py

```diff
@@ -0,0 +1,49 @@
+# ../players/engines/orangebox/cstrike.py
+
+"""Provides Team Fortress 2 specific Player based functionality."""
+
+# =============================================================================
+# >> IMPORTS
+# =============================================================================
+# Source.Python Imports
+from . import Player as _Player
+from configobj import ConfigObj
+from core import GAME_NAME
+from paths import SP_DATA_PATH
+
+# =============================================================================
+# >> CLASSES
+# =============================================================================
+class Player(_Player):
+    caching = True
+    
+    # Team
+    _team_data = ConfigObj(SP_DATA_PATH / 'teams' / GAME_NAME + '.ini', unrepr=True)
+
+    _teams_names = _team_data.get('names')
+    _class_names = _team_data.get('classes')
+
+    @property
+    def team_name(self):
+        teams_by_number = dict(zip(self._teams_names.values(), self._teams_names.keys()))
+        return teams_by_number[self.team]
+    
+    @team_name.setter
```

Comment:
Rather than having multiples `team_name` methods decorated as getter and setter I think it would be better to follow the overall design of the other properties across the whole project where they are distinctly available as methods as well. For example, `get_team_name` and `set_team_name` bound to the `team_name` property.